owncloudDomain/src/main/java/com/owncloud/android/domain/files/model/OCFile.kt

```diff
@@ -23,25 +23,52 @@ package com.owncloud.android.domain.files.model
 import android.os.Parcelable
 import android.webkit.MimeTypeMap
 import kotlinx.android.parcel.Parcelize
+import java.io.File
 import java.util.Locale
 
 //TODO: Add new attributes on demand. Let's try to perform a clean up :)
 @Parcelize
 data class OCFile(
-    val id: Long? = null,
-    val parentId: Long? = null,
+    var id: Long? = null,
+    var parentId: Long? = null,
     val owner: String,
-    val length: Long,
-    val creationTimestamp: Long? = null,
-    val modifiedTimestamp: Long,
+    var length: Long,
+    var creationTimestamp: Long? = 0,
+    var modificationTimestamp: Long,
     val remotePath: String,
-    val mimeType: String,
-    val etag: String? = null,
+    var mimeType: String,
+    var etag: String? = "",
     val permissions: String? = null,
-    val remoteId: String? = null,
-    val privateLink: String? = null
+    var remoteId: String? = null,
+    val privateLink: String? = "",
+    var storagePath: String? = null,
+    var treeEtag: String? = "",
+
+    // May not needed
```

Comment:
I separated must have and may have attributes. We will try to perform a cleanup. It will be fixed before merging new_arch/synchronization into master 👍